core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7739,3 +7743,70 @@ wasm_runtime_is_underlying_binary_freeable(WASMModuleCommon *const module)
 
     return true;
 }
+
+/*TODO: take us(below) out when have a linker */
+void
+wasm_runtime_release_imports(struct WasmExternInstance *extern_inst)
+{
+    if (extern_inst == NULL)
+        return;
+
+    wasm_runtime_free(extern_inst);
+}
+
+struct WasmExternInstance *
+wasm_runtime_create_imports(WASMModuleCommon *module,
+                            bool (*module_name_filter)(const char *))
+{
+    int32_t import_count = wasm_runtime_get_import_count(module);
+    struct WasmExternInstance *imports = NULL;
+
+    if (import_count == 0)
+        return NULL;
+
+    imports =
+        wasm_runtime_malloc(sizeof(struct WasmExternInstance) * import_count);
+    if (!imports) {
+        LOG_ERROR("allocate memory failed: %s", strerror(errno));
```

Comment:
errno and strerrno may be not supported in some platforms, and wasm_runtime_malloc may allocate memory from a global heap which uses wamr's ems heap allocator and there is no errno in it. Suggest to just `LOG_ERROR("allocate memory failed")`.